build_resolvers/lib/src/resolver.dart

```diff
@@ -282,13 +296,13 @@ class AnalyzerResolvers implements Resolvers {
   final Future<String> Function() _sdkSummaryGenerator;
 
   // Lazy, all access must be preceded by a call to `_ensureInitialized`.
-  AnalyzerResolver _resolver;
-  BuildAssetUriResolver _uriResolver;
+  late final AnalyzerResolver _resolver;
+  BuildAssetUriResolver? _uriResolver;
```

Comment:
It looks to me like this and `_packageConfig` could both be `late` and non-nullable as well?

Reply:
In `reset()`, we use `_uriResolver?.reset()` so we need to know whether it was initialized or not.  `_packageConfig` could be set in the constructor or in `_ensureInitialized`, so we need to check that too (and we can't move the initializer into a `late` field declaration because its async). I don't think we can actually avoid the null checks here?

Reply:
Ya if we need to do null checks then we need to keep this as you have it, sgtm